### DISCUSSION TITLE
How to store runtime data in json

### DISCUSSION THREAD

**Discussion by Ansuman92** [getgauge/taiko#1830] [General]
Hello Folks, Could anyone please let me know how to store runtime data in json and the same data can be used in another spec?

NOTE: We're using Gauge with Taiko in Javascript

Thanks in advance!

**Comment by zabil** [comment]
You can use data stores in Gauge to share data across scenarios/specs/suite. Documentation for this is available at
https://docs.gauge.org/writing-specifications.html?os=macos&language=javascript&ide=vscode#data-store

**Reply by Ansuman92** [reply]
Thanks for the quick reply. Have gone through the above documentation and looks like the data is being cleared post execution for all the three data store which are ScenarioStore, SpecStore and SuiteStore respectively.
Not sure whether the data created in one spec can be used as an input for another spec.
Please let me know if above point makes sense.

**Reply by sriv** [reply]
That does not sound right. A SuiteStore should preserve the data written till the suite execution completes. This should let your specs share data. 

Note that making specs share data could create an implicit coupling, which could lead to brittle tests. Specs should ideally be context free
